src/components/CodeBlock/normalize.ts

```diff
@@ -29,49 +29,55 @@ const stripComment = line =>
     ``,
   );
 
-const containsDirective = line =>
+const containsDirective = (line: string) =>
   [HIDE_DIRECTIVE, HIGHLIGHT_DIRECTIVE].some(expr => expr.test(line));
 
 /*
  * This parses the {1-3} syntax range that is sometimes used
  */
-const getInitialFilter = (className, split) => {
+const getInitialFilter = (className: string, split: string[]) => {
   const lineNumberExpr = /{([^}]+)/;
   const [, match] = className.match(lineNumberExpr) || [];
   if (match) {
     const lookup = match.split(/,\s*/).reduce((merged, range) => {
-      const [start, end = start] = range
+      const [start, end = start]: number[] = range
         .split(`-`)
         .map(num => parseInt(num, 10));
+
       for (let i = start; i <= end; i++) {
         merged[i - 1] = true;
       }
       return merged;
-    }, {});
+    }, {} as Record<number, boolean>);
```

Comment:
`as` remains

Reply:
Pretty sure you need this `as` otherwise the type is just empty object

Reply:
Reduce can take a generic which should cover this I think?

https://github.com/railwayapp/docs/pull/212/files#diff-ceb78c457ebfc2b44e9dd6896675ea4ea50f0ce4fec2d8071768f5bb71b50b05R166